airbyte-cdk/python/unit_tests/sources/test_availability.py

```diff
@@ -0,0 +1,193 @@
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.availability_strategy import AvailabilityStrategy, ScopedAvailabilityStrategy
+from airbyte_cdk.sources.streams.core import StreamData
+from airbyte_cdk.sources.streams.http.availability_strategy import HttpAvailabilityStrategy
+from airbyte_cdk.sources.streams.http.http import HttpStream
+
+logger = logging.getLogger("airbyte")
+
+
+class MockStream(Stream):
+    def __init__(self, name: str) -> Stream:
+        self._name = name
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def primary_key(self) -> Optional[Union[str, List[str], List[List[str]]]]:
+        pass
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[StreamData]:
+        pass
+
+
+class MockHttpStream(HttpStream):
+    url_base = "https://test_base_url.com"
+    primary_key = ""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.resp_counter = 1
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def path(self, **kwargs) -> str:
+        return ""
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        stub_resp = {"data": self.resp_counter}
+        self.resp_counter += 1
+        yield stub_resp
+    pass
+
+
+def test_no_availability_strategy():
+    stream_1 = MockStream("stream")
+    assert stream_1.availability_strategy is None
+
+    stream_1_is_available, _ = stream_1.check_availability(logger)
+    assert stream_1_is_available is True
+
+
+def test_availability_strategy():
+    class MockAvailabilityStrategy(AvailabilityStrategy):
+        def check_availability(self, stream: Stream, logger: logging.Logger, source: Optional[Source]) -> Tuple[bool, any]:
+            if stream.name == "available_stream":
+                return True, None
+            return False, f"Could not reach stream '{stream.name}'."
+
+    class MockStreamWithAvailabilityStrategy(MockStream):
+        @property
+        def availability_strategy(self) -> Optional["AvailabilityStrategy"]:
+            return MockAvailabilityStrategy()
+
+    stream_1 = MockStreamWithAvailabilityStrategy("available_stream")
+    stream_2 = MockStreamWithAvailabilityStrategy("unavailable_stream")
+
+    for stream in [stream_1, stream_2]:
+        assert isinstance(stream.availability_strategy, MockAvailabilityStrategy)
+
+    stream_1_is_available, _ = stream_1.check_availability(logger)
+    assert stream_1_is_available is True
+
+    stream_2_is_available, reason = stream_2.check_availability(logger)
+    assert stream_2_is_available is False
+    assert "Could not reach stream 'unavailable_stream'" in reason
+
+
+def test_scoped_availability_strategy():
```

Comment:
This test will be removed before merging. As we review PRs for connectors implementing availabilitystrategies, if we see enough common functionality, we can make a specific availabilitystrategy for scoped availability.